runtime/vm/ContinuationHelpers.cpp

```diff
@@ -535,11 +535,14 @@ acquireVThreadInspector(J9VMThread *currentThread, jobject thread, BOOLEAN spin)
 	/* Consistently re-fetch threadObj for all the cases below. */
 	threadObj = J9_JNI_UNWRAP_REFERENCE(thread);
 	vthreadInspectorCount = J9OBJECT_I64_LOAD(currentThread, threadObj, vm->virtualThreadInspectorCountOffset);
-	if (vthreadInspectorCount < 0) {
+
+	/* -1 indicates vthread is in transition and doesn't allow any inspector.
+	 * Count less than -1 indicates vthread was suspended in transition and every inspector acquired in this state
+	 * will continue to decrement the count by 1 to keep track of total inspector access granted.
+	 * Once the suspended vthread is resumed, no inspector should be allowed to acquire access before transition ends.
+	 */
+	if (vthreadInspectorCount < -1) {
```

Comment:
for `vthreadInspectorCount >= -1`, we will go into the `else if` block. If the +1 CAS operation succeeds, we will return true and allow the JVMTI operation when the virtual thread is IN_TRANSITION. the `else if` also needs to check for `vthreadInspectorCount >= 0`. for `vthreadInspectorCount == -1`, we should just retry. is this correct?